### PR TITLE
Validate function return type on GROUP BY

### PR DESCRIPTION
**Pull request by aelesbao**

### CODE REVIEW

**Comment by aelesbao** [comment]
@dobe could you review please?

**Comment by seut** [comment]
pls rebase

File: CHANGES.txt [hunk 1]
```diff
@@ -5,6 +5,8 @@ Changes for Crate
 Unreleased
 ==========
 
+ - Fixed an issue when grouping by a function with an invalid return type.
```
Comment (seut) [inline CHANGES.txt]: I'd suggest a more concrete entry like: Improved error handling when grouping by invalid data types

**Comment by seut** [comment]
lgtm

**Pull request merged**